tests/pytorch/distributed/test_comm_gemm_overlap.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+import os
+import subprocess
+from pathlib import Path
+
+import pytest
+import torch
+import transformer_engine.pytorch.cpp_extensions as tex
+from transformer_engine.pytorch.fp8 import FP8GlobalStateManager
+
+fp8_available, reason_for_no_fp8 = FP8GlobalStateManager.is_fp8_available()
+
+RNG_SEED: int = 1234
+NUM_PROCS: int = torch.cuda.device_count()
+SEQ_LENGTH: int = 2024
+BATCH_SIZE: int = 2
+NUM_HEADS: int = 64
+HEAD_DIM: int = 128
+HIDDEN_SIZE: int = NUM_HEADS * HEAD_DIM
```

Comment:
Can we get away with smaller model sizes?
```suggestion
RNG_SEED: int = 1234
NUM_PROCS: int = min(torch.cuda.device_count(), 4)
SEQ_LENGTH: int = 32 * NUM_PROCS
BATCH_SIZE: int = 1
NUM_HEADS: int = 1
HEAD_DIM: int = 32 * NUM_PROCS
HIDDEN_SIZE: int = NUM_HEADS * HEAD_DIM
```
We have separate tests to check the numerical correctness of TE modules, so these tests should just focus on communication. Smaller models also run faster and we can apply tighter tolerances.

Also, it doesn't seem that these values are passed to the run scripts.

Reply:
I'm adding these to the test cmd and we can make them smaller too but the GEMM op itself is quite fast even at these sizes. The limiting factor in this case is the `torch.distributed` launch, and that's independent of problem sizing. I've tried to do the distributed launch from within the pytest script via `torch.multiprocessing` but it turned out to be incompatible with device-to-device comms (it only works for simpler distributed cases like pure data-parallel). So at least for now, we're stuck with the overhead of separate `torchrun` calls.

That said, I do really have to check numerical correctness here to make sure the comms aren't mangling the order of the work chunks.